cmd/feed-ingress/main.go

```diff
@@ -40,6 +40,8 @@ var (
 	pushgatewayIntervalSeconds     int
 	pushgatewayLabels              cmd.KeyValues
 	controllerConfig               controller.Config
+	mainIngress                    bool
```

Comment:
I think it might be clearer to call this variable `defaultIngress` and have `ingressName` default to `main` so that as a user, by default this feed instance can act as the default controller and attach to the main elb pair.

Reply:
> have `ingressName` default to `main` so that as a user, by default this feed instance can act as the default controller and attach to the main elb pair.

I thought about this but decided it was safer to not default to main.  The reason is that if a test instance is deployed without this configuration it would then attach to the main ELBs and potentially break the main ingress.  What do you think?

Reply:
> couldn't we have an ingress without a name? and avoid this main boolean?
I am just unsure as of why do we need it. If we want to maintain backwards compatibility, a nameless feed would be a way to have it, and if you want specific ingresses, you name them

The three of us discussed the two approaches:
1. The current implementation of a 'main' feed that handles ingresses that don't have an ingress name specified for backwards compatibility
- requires the main elbs to be tagged and feed-ingress can only attach if it's given the tag value
- this is a safety feature to ensure that test feed-ingress pods that haven't been configured don't attach to the main elbs
- requires an extra bool argument so that the main feed-ingress instance know to handle ingresses without a name, for backwards compatibility
2. An ingress that doesn't have a name that will be the main ingress
- don't add the ingress-name tag to the main elb and never add ingress-name to ingresses that are for the main / default elb
- doesn't require the main-ingress bool to handle two cases
- may need to update the logic that attaches to the elb to handle the no tag case
- potential risk that test pods without ingress-name specified could attach to the main elb
- could mitigate this by e.g. requiring the ingress-name arg and checking that it's not the same as a default randomly generated string (although, there's still a chance that someone will set a required arg to an empty string if they are just playing around :)

Reply:
>* don't add the ingress-name tag to the main elb ...
>* may need to update the logic that attaches to the elb to handle the no tag case

we also said that we could intentionally add the tag our current elb with an empty name, so that the logic needs no change. I think an empty string is valid (unnamed). 

> * (although, there's still a chance that someone will set a required arg to an empty string if they are just playing around :)

given that the params should be given as either  `--main-ingress` (option 1) or `--ingress-name ""` (option 2), I'd say that there are more chances of people accidentally using --main-ingress.

Reply:
`--main-ingress` defaults to false so this would need to be given as `--main-ingress=true` (or one of the other variants of true) so you would have to intentionally being setting it.  Still not fool proof so not ideal.

Reply:
I think flags are not really the mechanism we would use to prevent one feed-ingress attaching to an incorrect elb by miss-configuration. It would probably be more on the side of "does this instance of feed has credentials with permissions to attach itself to this elb?"

Reply:
In our case, test elbs can't actually attach to the main elb because they get a different role.  But that's just how we've done it (recently) and not provided by feed.

Reply:
Ah great! In my opinion this should be that safety mechanism :)

Reply:
I am of the opinion that the boolean field may not be necessary. This is because we are anyways advertising this as a breaking change, which means we can lose backwards compatibility. I would suggest for the feed ingress to process only those entries with the name annotation matching the name of the ingress and ignore anything which does not contain the name. 

It would still be a seamless change for the application teams at least in our case because we can make the change in k8plugin to have a name defaulted to the main ingress (which is the equivalent of the current breed of ingress we have). There can be a cut-off date before which all the teams should specify the ingress name. 

I am not fully convinced that we need to protect the test app getting attached to the main ingress/ELB. Such a mechanism does not exist now and also there's nothing stopping from the test attaching to the main ingress by specifying the name of the main ingress. Such mistakes can be caught via monitoring and alerting and am not convinced we need a boolean flag to protect. 

I would simply make the name mandatory for the ingress to start and a corresponding ELB with a tag matching the ingress name, without which the ingress should not get to running state.